helix-core/src/test/java/org/apache/helix/integration/rebalancer/DelayedAutoRebalancer/TestDelayedAutoRebalanceWithDisabledInstance.java

```diff
@@ -292,6 +292,85 @@ public void testDisableDelayRebalanceInInstance() throws Exception {
     super.testDisableDelayRebalanceInInstance();
   }
 
+  @Test(dependsOnMethods = {"testDisableDelayRebalanceInInstance"})
+  @Override
```

Comment:
Following tests are bit duplicated. We already test on demand rebalance and test filtering out hosts in previous tests.

Reply:
`testOnDemandRebalance` in `TestDelayedAutoRebalance.java` tests only offline nodes. `testOnDemandRebalance` in `TestDelayedAutoRebalanceWithDisabledInstance.java` is for disabled instances.

Reply:
Or do you think it would be better to factor that out? Well..I personally feel like not factoring this out could be more intuitive. And In the future if you have more features you wanna to `disabled` nodes only or `offline` nodes only, then you can modify the test accordingly without touching the counterparts (And I believe it's gonna happen shortly when you implementing node swag). Just my two cents. And we can sync up offline today. ^_^

Reply:
The change we made in this PR is how we filter out instances for further stages in pipeline.  Since no changes are made in further pipeline, we don't need to add test for each rebalance flavor.
We do need to test when there are disabled or offline instances. We could test them in the same unit test. Having duplicate code is not a good practice.

Reply:
As discussed offline, removed the test cases for rack aware related classes and factored out the code.